File: r/src/get_token.R
library(httr2)
library(jsonlite)
library(dotenv)
library(glue)

# Load environment variables
load_dot_env()

BASE_URL <- "https://api.hyperswitchai.com"

main <- function() {
  tryCatch({
    # Check for required environment variables
    if (Sys.getenv("USERNAME") == "" || Sys.getenv("PASSWORD") == "") {
      stop("USERNAME and PASSWORD must be set in a .env file in the r directory. You can rename .env.sample to .env and fill in your own email and password.")
    }

    # Get token
    response <- request(glue("{BASE_URL}/auth")) %>%
      req_headers("Content-Type" = "application/json") %>%
      req_body_json(list(
        username = Sys.getenv("USERNAME"),
        password = Sys.getenv("PASSWORD")
      )) %>%
      req_perform()

    if (resp_status(response) != 200) {
      stop("Authentication failed")
    }

    result <- resp_body_json(response)
    
    # Print success message with green color and token only once
    cat("\n\033[32m✅ Token successfully retrieved\033[0m\n")
    cat("\nToken:", result$token, "\n")  # Print token only once

  }, error = function(e) {
    cat("Error:", e$message, "\n")
  })
}

# Execute if run directly
if (sys.nframe() == 0) {
  main()
} 

File: rust/src/bin/get_token.rs
use anyhow::Result;
use colored::Colorize;
use std::env;

const BASE_URL: &str = "https://api.hyperswitchai.com";

#[derive(serde::Serialize)]
struct AuthRequest {
    username: String,
    password: String,
}

#[derive(serde::Deserialize)]
struct AuthResponse {
    token: String,
}

#[tokio::main]
async fn main() -> Result<()> {
    dotenv::dotenv()?;

    // Check for required environment variables
    if env::var("USERNAME").is_err() || env::var("PASSWORD").is_err() {
        anyhow::bail!("USERNAME and PASSWORD must be set in a .env file in the rust directory. You can rename .env.sample to .env and fill in your own email and password.");
    }

    let auth_req = AuthRequest {
        username: env::var("USERNAME")?,
        password: env::var("PASSWORD")?,
    };

    let client = reqwest::Client::new();
    let response = client
        .post(format!("{}/auth", BASE_URL))
        .json(&auth_req)
        .send()
        .await?;

    if !response.status().is_success() {
        anyhow::bail!("Authentication failed: {}", response.status());
    }

    let auth_resp: AuthResponse = response.json().await?;

    println!("\n{}", "✅ Token successfully retrieved".green());
    println!("Token: {}", auth_resp.token);

    Ok(())
} 